src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -4752,27 +4752,30 @@ int wm_vuldet_set_agents_info(agent_software **agents_software, update_node **up
             agent_dist = FEED_WIN;
         } else if (strcasestr(os_platform, vu_feed_ext[FEED_MAC])) {
             char *version = NULL;
+            char *save_ptr = NULL;
             w_strdup(agent_version, version);
 
             if (agent_version) {
-                strtok(agent_version, "v");
-                agent_major=strtok(NULL, ".");
-                agent_minor=strtok(NULL, ".");
-                if (agent_major == NULL) {
-                    mdebug1("Cannot parse agent_major from version agent");
-                }
-                if (agent_minor == NULL) {
-                    mdebug1("Cannot parse agent_minor from version agent");
+                strtok_r(agent_version, "v", &save_ptr);
+                agent_major = strtok_r(NULL, ".", &save_ptr);
+                agent_minor = strtok_r(NULL, ".", &save_ptr);
+                if (!agent_major || !agent_minor) {
+                    dist_error = -2;
+                    mterror(WM_VULNDETECTOR_LOGTAG, VU_VER_READING_ERROR, id);
+                } else {
+                    if (atoi(agent_major) > 4 || (atoi(agent_major) == 4 && atoi(agent_minor) > 0)) {
+                        agent_dist_ver = FEED_MAC;
+                        agent_dist = FEED_MAC;
+                    } else {
+                        dist_error = -2;
+                        mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_AGENT_UNSOPPORTED, id, version);
+                    }
                 }
-            }
-
-            if (atoi(agent_major) > 4 || (atoi(agent_major) == 4 && atoi(agent_minor) > 0)) {
-                agent_dist_ver = FEED_MAC;
-                agent_dist = FEED_MAC;
             } else {
                 dist_error = -2;
-                mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_AGENT_UNSOPPORTED, id, version);
+                mterror(WM_VULNDETECTOR_LOGTAG, VU_VER_READING_ERROR, id);
```

Comment:
It is better to distinguish this error message from the one on line 4764. They are exactly the same and it makes the debugging harder.